src/openforms/js/components/admin/form_design/logic/actions/Actions.js

```diff
@@ -105,28 +107,62 @@ const ActionFetchFromService = ({action, errors, onChange}) => {
     setIsModalOpen(false);
   };
 
+  const formContext = useContext(FormContext);
+
+  let serviceFetchConfigFromVar =
+    _.cloneDeep(formContext.formVariables.find(element => element.key === action.variable))
```

Comment:
this should also filter the variables down to user defined variables like the dropdown/variable selection, no?

Reply:
No, this gets the ServiceFetchConfig from the variable set on the action passed in. The configs are an attribute on the variable of the action... The initial release, the action could change any compontent value and the backend created a faux variable, just to evaluate the fetch action. That's changed now, as the configs are now actually bound to a user defined variable as originally specced out... So yes; filtering is done elsewhere to only allow picking of user defined variables such that `action.variable` here is always in that set.

:thinking: I need to test that case when writing #2884: in the current release create a fetch action that sets the choices of a selectbox or other component directly, and see how it migrates and behaves in the new release.

Reply:
I think I understand :sweat_smile: